autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/SecureStoreBackedAutofillStore.kt

```diff
@@ -340,11 +355,12 @@ class SecureStoreBackedAutofillStore @Inject constructor(
         }
     }
 
-    private fun usernameMatch(
+    private suspend fun usernameMatch(
         credentials: WebsiteLoginDetailsWithCredentials,
         username: String?,
     ): Boolean {
-        return credentials.details.username != null && credentials.details.username == username
+        // special case where we don't want to consider two null usernames as equals
+        return credentials.details.username != null && usernameComparer.isEqual(credentials.details.username, username)
```

Comment:
use `usernameComparer` instead of a direct string comparison

note, here we also need to preserve the previous code’s special case where we don’t want to consider two null usernames as equals here